config-model/src/main/java/com/yahoo/vespa/model/admin/clustercontroller/ClusterControllerContainer.java

```diff
@@ -110,7 +110,10 @@ public void getConfig(ZookeeperServerConfig.Builder builder) {
 
     @Override
     public void getConfig(QrStartConfig.Builder builder) {
-        builder.jvm(new QrStartConfig.Jvm.Builder().heapsize(512));
+        builder.jvm.heapsize(512);
+        builder.jvm.heapSizeAsPercentageOfPhysicalMemory(0);
```

Comment:
It should not be necessary to set the same value as the default in the def file. We should instead avoid setting the wrong values at higher levels in the model. The handling of `getMemoryPercentage` in `ContainerCluster.getConfig(QrStartConfig)` should be moved to the `ApplicationContainerCluster` class. Only those clusters are created by `ContainerModelBuilder`, and that's the only place where we set `memoryPercentage`.

Reply:
The intention is not to set it to the same as default config. If you set heapSize you need to ensure that heapSizeAsPercentageOfPhysicalMemory is 0. The fact that it is 0 in default config can change. Now that does not matter.

I was not aware of ApplicationContainerCluster, at least not when memory percentage stuff was written. I will move it there, but as a separate PR.

Reply:
True. But then we should perhaps set it to 0 in ContainerCluster's getConfig() and override it only in ApplicationContainerCluster.

Reply:
Yes, I will go over and consolidate QrStartConfig generation in the next PR.